src/dbnode/storage/repair.go

```diff
@@ -274,15 +282,15 @@ func (r shardRepairer) Repair(
 		session := sessions[i].session
 		perSeriesReplicaIter, err := session.FetchBlocksFromPeers(nsMeta, shard.ID(), level, metadatasToFetchBlocksFor, rsOpts)
 		if err != nil {
-			return repair.MetadataComparisonResult{}, err
+			return metadataRes, err
```

Comment:
Why the change here? It's a pretty common Go pattern to return nil(-equivalent) values when returning an error.

Reply:
I am changing it back to nil. Although the thought process behind this change was that we are throwing away work done to compare so its better to return with an error and leave it to the caller to decide if they want to log result even in case of an error.